tobac/utils/bulk_statistics.py

```diff
@@ -301,16 +307,14 @@ def get_statistics_from_mask(
             field.sel(time=tt).values if "time" in field.coords else field.values
             for field in fields
         )
+
         features_t = features.loc[features.time == tt].copy()
 
         # make sure that the labels in the segmentation mask exist in feature dataframe
-        if (
-            np.intersect1d(np.unique(segmentation_mask_t), features_t[id_column]).size
-            > np.unique(segmentation_mask_t).size
-        ):
-            raise ValueError(
-                "The labels of the segmentation mask and the feature dataframe do not seem to match. Please make sure you provide the correct input feature dataframe to calculate the bulk statistics. "
-            )
+        # continue loop because not all timesteps might have matching IDs
+        if not np.any(np.isin(features_t[id_column], np.unique(segmentation_mask_t))):
+            warnings.warn("Not all timesteps have matching features", UserWarning)
+            continue
```

Comment:
This may results in timesteps without matching features/segments missing from the output dataframe. We should append the input features as is and hope pd.concat sorts it out.

Reply:
addressed this in latest commit :)